test/parallel/test-assert.js

```diff
@@ -645,6 +645,24 @@ try {
                      'Message incorrectly marked as generated');
 }
 
+// verify custom errors
+try {
+  assert.strictEqual(1, 2, new RangeError('my range'));
+} catch (e) {
+  assert.strictEqual(e.message, 'my range');
+  assert.ok(e instanceof RangeError, 'Incorrect error type thrown');
+}
+
+// verify AssertionError is the result from doesNotThrow with custom Error
+try {
+  assert.doesNotThrow(() => {
+    throw new TypeError('wrong type');
+  }, TypeError, new RangeError('my range'));
+} catch (e) {
+  assert.ok(e.message.includes('my range'));
+  assert.ok(e instanceof assert.AssertionError);
```

Comment:
For these tests involving `try...catch`, we don't know if the assertions were actually run. Can you add checks like [these](https://github.com/geek/node/blob/8ec90de7ccada7073aeae22e1dc69037bf1da11f/test/parallel/test-assert.js#L619-L629)